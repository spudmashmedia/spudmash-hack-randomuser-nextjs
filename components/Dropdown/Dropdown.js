/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React from "react";
import { FormControl, InputLabel, Select } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

let useStyles = makeStyles(theme => ({
  formcontrol: {
    margin: theme.spacing(1),
    width: 200
  }
}));

let Dropdown = props => {
  let classes = useStyles();
  let label = props.label;
  let data = props.data;
  let onChangeHandler = props.onChangeHandler;

  return (
    <FormControl className={classes.formcontrol} noValidate autoComplete="off">
      <InputLabel>{label}</InputLabel>
      <Select native onChange={e => onChangeHandler(e, e.target.value)}>
        {data.map(x => (
          <option key={x.id} value={x.id}>
            {x.label}
          </option>
        ))}
      </Select>
    </FormControl>
  );
};

export default Dropdown;
