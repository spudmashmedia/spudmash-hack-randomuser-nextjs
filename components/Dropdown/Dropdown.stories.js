/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React, { useState } from "react";
import {
  Paper,
  Typography,
  Card,
  CardHeader,
  CardContent
} from "@material-ui/core";
import CheckBox from "@material-ui/icons/CheckBox";
import Dropdown from "./Dropdown";
import * as constants from "../../common/constants";
import { withA11y } from '@storybook/addon-a11y';

export default {
  component: Dropdown,
  title: "Dropdown",
  decorators: [withA11y],
  excludeStories: /.*Data$/
};

export const Default = () => {
  const [myGender, setMyGender] = useState(constants.gender.map(x => x.id)[0]);
  const [myCountry, setMyCountry] = useState(
    constants.countryCodes.map(x => x.id)[0]
  );

  let handleMyGender = (event, newValue) => {
    setMyGender(newValue);
    console.log(myGender);
  };
  let handleMyCountry = (event, newValue) => {
    setMyCountry(newValue);
    console.log(myCountry);
  };

  return (
    <div>
      <Paper>
        <Card>
          <CardContent>
            <Dropdown
              name="test-dropdown"
              label="Test Dropdown"
              data={constants.gender}
              onChangeHandler={handleMyGender}
            />
            <Dropdown
              name="test-dropdown"
              label="Test Dropdown"
              data={constants.countryCodes}
              onChangeHandler={handleMyCountry}
            />
          </CardContent>
        </Card>
        <Card>
          <CardHeader avatar={<CheckBox />} title="Output" />
          <CardContent>
            <Typography>Selected gender: {myGender}</Typography>
            <Typography>Selected country code: {myCountry}</Typography>
          </CardContent>
        </Card>
      </Paper>
    </div>
  );
};
