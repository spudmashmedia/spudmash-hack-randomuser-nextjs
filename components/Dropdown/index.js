/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import Dropdown from "./Dropdown";
export default Dropdown;
