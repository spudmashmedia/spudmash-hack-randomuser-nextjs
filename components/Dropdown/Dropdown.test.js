/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Spudmash Media Pty Ltd
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

import React from "react";
import Dropdown from "./Dropdown";
import renderer from "react-test-renderer";

describe("ErrorWidget Component Snapshot Test", () => {
  test("renders correctly", () => {
    const actionData = [
      { id: "male", label: "Male" },
      { id: "female", label: "Female" }
    ];

    const mockOnChangeHandler = jest.fn((e, newValue) => "male");

    const tree = renderer
      .create(
        <Dropdown
          name="test-dropdown"
          label="Test Dropdown"
          data={actionData}
          onChangeHandler={mockOnChangeHandler}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
