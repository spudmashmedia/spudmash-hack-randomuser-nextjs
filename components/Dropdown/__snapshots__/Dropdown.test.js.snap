// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorWidget Component Snapshot Test renders correctly 1`] = `
<div
  autoComplete="off"
  className="MuiFormControl-root makeStyles-formcontrol-1"
  noValidate={true}
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink={false}
  >
    Test Dropdown
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <select
      aria-invalid={false}
      className="MuiSelect-root MuiSelect-select MuiInputBase-input MuiInput-input"
      disabled={false}
      multiple={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <option
        value="male"
      >
        Male
      </option>
      <option
        value="female"
      >
        Female
      </option>
    </select>
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
